### PR TITLE
lunar-client: init at version 2.4.0

### PR DESCRIPTION
**Pull request by zyansheep**
###### Motivation for this change
Lunar Client is a pretty popular minecraft client / launcher

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
^^^ This is on by default on NixOS right?
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/lunar-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    categories = "Game;";
+  };
+  appimageContents = appimageTools.extract {
+    inherit name src;
+  };
+  src = fetchurl {
+    url = "https://launcherupdates.lunarclientcdn.com/Lunar%20Client-${version}.AppImage";
+    name = "lunar-client.AppImage";
+    sha256 = "bb85a62127a9b3848cc60796c20ac75655794f1d3cd17cb6b5499bbf19d16019";
+  };
```
Comment (SuperSandro2000) [inline pkgs/games/lunar-client/default.nix:22]:
```suggestion
  version = "2.4.0";

  desktopItem = makeDesktopItem {
    name = "Lunar Client";
    exec = "lunar-client";
    icon = "lunarclient";
    comment = "Optimized Minecraft Client for 1.7.10 and 1.8.9";
    desktopName = "Lunar Client";
    genericName = "Minecraft Client";
    categories = "Game;";
  };

  appimageContents = appimageTools.extract {
    inherit name src;
  };

  src = fetchurl {
    url = "https://launcherupdates.lunarclientcdn.com/Lunar%20Client-${version}.AppImage";
    name = "lunar-client.AppImage";
    sha256 = "bb85a62127a9b3848cc60796c20ac75655794f1d3cd17cb6b5499bbf19d16019";
  };
```

File: pkgs/games/lunar-client/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+in appimageTools.wrapType1 rec {
+  inherit name src;
+
+  extraInstallCommands = ''
+    mkdir -p $out/share/applications
+    cp ${desktopItem}/share/applications/* $out/share/applications
+    cp -r ${appimageContents}/usr/share/icons/ $out/share/
+  '';
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/games/lunar-client/default.nix:31]:
```suggestion
  '';

  meta = with lib; {
```

File: pkgs/games/lunar-client/default.nix (hunk 1 above)
Comment (zyansheep) [reply pkgs/games/lunar-client/default.nix:22]: I respect a person who likes their spacing : )

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Please fix the commit naming and history. The last 3 fix-ups should be squashed into the package init commit and the entry in the maintainer-list is usually a stand-alone commit before your first packaging commit.

```bash
git reset --soft HEAD~5 # throws away last 5 commits' information without resetting file changes
git reset HEAD pkgs/games/lunar-client/default.nix # remove lunar-client from staging area
git commit # "maintainers: add zyansheep"
git add pkgs/games/lunar-client/default.nix
git commit # "lunar-client: init at 2.4.0"

git log
# when you think the git history looks good
git push -f
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -5112,6 +5112,8 @@ in
 
   ltwheelconf = callPackage ../applications/misc/ltwheelconf { };
 
+  lunar-client = callPackage ../games/lunar-client {};
+
```
Comment (OPNA2608) [inline pkgs/top-level/all-packages.nix:5116]: Whoops, this needs to go in the lunar-client commit as well - together with the actual package code. Terribly sorry for missing this! :sweat:

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
I'll test the build & binary after work today, will report back.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (zyansheep) [reply pkgs/top-level/all-packages.nix:5116]: nah, I totally missed that too!

**Review by OPNA2608 — APPROVED** [review]
LGTM.

Result of `nixpkgs-review pr 103229` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lunar-client</li>
  </ul>
</details>

**Comment by zyansheep** [comment]
What else needs to be done to get this merged?

**Comment by OPNA2608** [comment]
Someone with merge permissions needs to see this PR and merge it, that's it really. Takes abit of patience and persistence to find someone sometimes, you can try posting a link to your reviewed PR [here](https://discourse.nixos.org/t/prs-already-reviewed/).

**Comment by zyansheep** [comment]
Hmm... Seems like there needs to be a better way to add packages then to rely on a small group of people to vet each and every one. Perhaps something like the Arch User Repository is needed? Maybe I'll open an rfc...

**Comment by OPNA2608** [comment]
> Perhaps something like the Arch User Repository is needed? Maybe I'll open an rfc...

You're looking for the [NUR](https://github.com/nix-community/NUR) then.

**Comment by AndersonTorres** [comment]
Looks unfree to me, looks good to Borg :)

**Pull request merged**